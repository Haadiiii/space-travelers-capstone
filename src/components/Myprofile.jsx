import React from "react";
import "./myprofile.css";
import { useSelector } from "react-redux";

export default function Myprofile() {
  const reservedRockets = useSelector((state) => state.rocket.filter((item) => item.reserved));
  const reservedMission = useSelector((state) => state.mission).filter((item) => item.reserved);

  return (
    <div className="profile-content">
      <section className="reserved-rockets">
        <h1 className="heading">My Rockets</h1>
        {reservedRockets.map((rocket) => (
          <ul key={rocket.id} className="reserved-table">
            <li>{rocket.name}</li>
          </ul>
        ))}
      </section>
      <section>
        <h1 className="heading">My Missions</h1>
        {reservedMission.map((mission) => (
          <ul key={mission.id} className="reserved-table">
            <li>{mission.name}</li>
          </ul>
        ))}
      </section>
    </div>
  );
}
