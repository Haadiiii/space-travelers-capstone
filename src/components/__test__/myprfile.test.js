import React from "react";
import renderer from "react-test-renderer";
import Myprofile from "./__mock__/myprofile";

it('This is a test to check the My Profile page', () => {
  const tree = renderer.create(
    <Myprofile />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
