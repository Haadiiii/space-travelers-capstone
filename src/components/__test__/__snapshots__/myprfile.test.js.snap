// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`This is a test to check the My Profile page 1`] = `
<div
  className="profile-content"
>
  <section
    className="reserved-rockets"
  >
    <h1>
      My Rockets
    </h1>
    <table>
      <tbody>
        <tr>
          <td>
            rocket
          </td>
        </tr>
      </tbody>
    </table>
  </section>
  <section>
    <h1>
      My Missions
    </h1>
    <table>
      <tbody>
        <tr>
          <td>
            mission
          </td>
        </tr>
      </tbody>
    </table>
  </section>
</div>
`;
